Spigot-API-Patches/0157-Turtle-API.patch

```diff
@@ -0,0 +1,278 @@
++
++    public TurtleLayEggEvent(Turtle turtle, Location location, int eggCount) {
++        super(turtle);
++        this.location = location;
++        this.eggCount = eggCount;
++    }
++
++    /**
++     * The turtle going home
++     *
++     * @return The turtle
++     */
++    public Turtle getEntity() {
++        return (Turtle) entity;
++    }
++
++    /**
++     * Get the location where the eggs are being laid
++     *
++     * @return Location of eggs
++     */
++    public Location getLocation() {
++        return location;
++    }
++
++    /**
++     * Get the number of eggs being laid
++     *
++     * @return Number of eggs
++     */
++    public int getEggCount() {
++        return eggCount;
++    }
++
++    /**
++     * Set the number of eggs being laid
++     *
++     * @param eggCount Number of eggs
++     */
++    public void setEggCount(int eggCount) {
++        if (eggCount < 1) {
++            cancelled = true;
++            return;
++        }
++        eggCount = Math.min(eggCount, 4);
++    }
++
++    @Override
++    public boolean isCancelled() {
++        return cancelled;
++    }
++
++    @Override
++    public void setCancelled(boolean cancel) {
++        cancelled = cancel;
++    }
++
++    public HandlerList getHandlers() {
++        return handlers;
++    }
++
++    public static HandlerList getHandlerList() {
++        return handlers;
++    }
++}
+diff --git a/src/main/java/com/destroystokyo/paper/event/entity/TurtleStartDiggingEvent.java b/src/main/java/com/destroystokyo/paper/event/entity/TurtleStartDiggingEvent.java
+new file mode 100644
+index 00000000..eaf7f51d
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/entity/TurtleStartDiggingEvent.java
+@@ -0,0 +1,65 @@
++package com.destroystokyo.paper.event.entity;
++
++import org.bukkit.Location;
++import org.bukkit.entity.Turtle;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++
++/**
++ * Fired when a Turtle starts digging to lay eggs
++ */
++public class TurtleStartDiggingEvent extends EntityEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++    private boolean cancelled = false;
++    private final Location location;
++
++    public TurtleStartDiggingEvent(Turtle turtle, Location location) {
++        super(turtle);
++        this.location = location;
++    }
++
++    /**
++     * The turtle
++     *
++     * @return The turtle
++     */
++    public Turtle getEntity() {
++        return (Turtle) entity;
++    }
++
++    /**
++     * Get the location where the turtle is digging
++     *
++     * @return Location where digging
++     */
++    public Location getLocation() {
++        return location;
++    }
++
++    public void setEggCount(int eggCount) {
++        if (eggCount < 1) {
++            cancelled = true;
++            return;
++        }
++        eggCount = Math.min(eggCount, 4);
```

Comment:
In my opinion this maximum should be mentioned in the documentation.

Reply:
this shouldn't have been there. removed.